File: app/src/at-myhuynh/java/asiantech/internship/summer/java_core/ex16/ShowNumberExist.java
package asiantech.internship.summer.java_core.ex16;

import java.util.ArrayList;
import java.util.List;
import java.util.logging.Level;
import java.util.logging.Logger;

import asiantech.internship.summer.java_core.common.Common;

public class ShowNumberExist {

    public static void main(String[] args) {
        List<Integer> listNumber = new ArrayList<>();
        int n = Common.input("Nhập vào độ dài mảng: ");
        for (int i = 0; i < n; i++) {
            listNumber.add(Common.input("Nhập phần tử thứ " + i + ": "));
        }

        showNumberExist(listNumber);
    }

    private static void showNumberExist(List<Integer> listNumber) {
        Logger log = Logger.getLogger(ShowNumberExist.class.getName());
        for (int i = 0; i < listNumber.size(); i++) {
            if (isExist(listNumber.get(i), listNumber, i)) {
                if (log.isLoggable(Level.INFO)) {
                    log.info(listNumber.get(i) + ": " + countNumberExist(listNumber.get(i), listNumber, i));
                }
            }
        }
    }

    private static int countNumberExist(int n, List<Integer> listNumber, int index) {
        int count = 1;
        for (int i = index + 1; i < listNumber.size(); i++) {
            if (n == listNumber.get(i) && isExist(n, listNumber, index)) {
                count++;
            }
        }
        return count;
    }

    private static boolean isExist(int n, List<Integer> listNumber, int index) {
        for (int i = 0; i < index; i++) {
            if (n == listNumber.get(i)) {
                return false;
            }
        }
        return true;
    }

}

File: app/src/at-myhuynh/java/asiantech/internship/summer/kotlin/ex16/ShowNumberExist.kt
package asiantech.internship.summer.kotlin.ex16

import asiantech.internship.summer.kotlin.common.Common
import java.util.ArrayList

object ShowNumberExist {

    @JvmStatic
    fun main(args: Array<String>) {
        val listNumber = ArrayList<Int>()
        val n = Common.input("Nhập vào độ dài mảng: ")
        for (i in 0 until n) {
            listNumber.add(Common.input("Nhập phần tử thứ $i: "))
        }

        showNumberExist(listNumber)
    }

    private fun showNumberExist(listNumber: List<Int>) {
        for (i in listNumber.indices) {
            if (isExist(listNumber[i], listNumber, i)) {
                println("${listNumber[i]} : " + countNumberExist(listNumber[i], listNumber, i))
            }
        }
    }


    private fun countNumberExist(n: Int, listNumber: List<Int>, index: Int): Int {
        var count = 1
        for (i in index + 1 until listNumber.size) {
            if (n == listNumber[i] && isExist(n, listNumber, index)) {
                count++
            }
        }
        return count
    }

    private fun isExist(n: Int, listNumber: List<Int>, index: Int): Boolean {
        for (i in 0 until index) {
            if (n == listNumber[i]) {
                return false
            }
        }
        return true
    }
}
